server/expression.js

```diff
@@ -61,22 +78,33 @@ expression.query.Matrix;
 /**
  * @typedef {{
  *   matrixName: string,
- *   gene: string
+ *   geneNames: !Array<string>,
+ *   conditionNames: !Array<string>
  * }}
  */
 expression.query.Profile;
 
 // Start public APIs
 /**
+ * @param {!expression.query.MatrixAll} query
+ * @param {string} expressionPath
+ * @return {?expression.MatrixAll}
+ */
+expression.query.matrixAll = function(query, expressionPath) {
+  var file = expressionPath + query.matrixName;
+  return expression.readExpressionAll_(file);
```

Comment:
It seems that purpose is to get only rowNames, colNames and value min/max of a given matrix. The query name here is confusing, as it indicates that you want to get the entire matrix (including their values). 

Maybe you can modify the matrix query, to indicate that you don't need values to be sent back. That could save an additional query type.